inject: restore City of Khamoon mummy

lahm86 opened the pull request:
Resolves #886.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This option places the mummy and its triggers as follows in City of Khamoon (triggered both in the flipped and unflipped rooms). So the mummy spawns by the switch, and is triggered on the tile below the pushblock.

![image](https://github.com/LostArtefacts/Tomb1Main/assets/33758420/d1ae6796-1e16-41e3-a7da-9d68018e6a4b)

I had to move `Stats_ObserveRoomsLoad` in `level.c` to take place post-injection for the enemy kill count to update correctly. While this wasn't a problem previously for the restored bat in Colosseum, it is here because we're changing the `index` on two sectors that don't currently have any floor data. So as a result, the kill count remains correct.

I've ticked off `fix floor data issues` in the readme for now as this is the final topic discussed in #755.

lahm86 committed:
inject: restore City of Khamoon mummy

Added an option to restore the PS mummy in City of Khamoon room 25.
This required moving the stats room caching call to occur after
injection, otherwise inserted triggers aren't detected and hence the
kill count would be wrong.

Resolves #886.

lahm86 commented:
@ajtudela Could you provide translations for this new option? Thanks.

rr- reviewed (commented):
Can we make this option more generic, such as "restore PS1 enemies" and then only in the option description talk about the mummy in specific?

lahm86 committed:
make option more general

lahm86 commented:
> Can we make this option more generic, such as "restore PS1 enemies" and then only in the option description talk about the mummy in specific?

Updated now, and I updated the vars/enums as well to be consistent. LMK what you think.

![image](https://github.com/LostArtefacts/Tomb1Main/assets/33758420/054e7e6c-3655-4cbc-aeb8-19015c3d62d6)

walkawayy commented on CHANGELOG.md:
```diff
@@ -2,6 +2,7 @@
 - added the current music track and timestamp to the savegame so they now persist on load (#419)
 - added the triggered music tracks to the savegame so one shot tracks don't replay on load (#371)
 - added forward/backward input detection in line with TR2+ for jump-twists (#931)
+- added an option to restore the mummy in City of Khamoon room 25, similar to the PS version (#886)
```
Idc personally but all other changelog entries use `PS1`. Maybe we should be consistent? Or we could retroactively change them all to `PS` or `PSX`. Whatever @rr- prefers.

rr- replied:
I'm fine with both, props for consistency

lahm86 replied:
Changed to PS1 to match.

walkawayy commented on tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json:
```diff
@@ -239,6 +239,10 @@
     },
+    "restore_ps1_enemies": {
+      "Title": "Restore PS1 enemies",
+      "Description": "Adds the mummy that appears in the PlayStation version of City of Khamoon, room 25.\nNB Changing this option will require restarting the level."
```
What does `NB` mean?

lahm86 replied:
Nota bene - maybe too formal?

rr- replied:
Let's just remove it, as it doesn't add much value

lahm86 replied:
NB: NB removed :)

walkawayy commented:
Also going to be annoying and link #352 since I believe that is also a floor data issue.

lahm86 commented:
> Also going to be annoying and link #352 since I believe that is also a floor data issue.

I'm not convinced this is a floor data issue. The trigger there has a timer value of 2, and changing this makes no difference to the outcome. I think there is a bug somewhere in the camera control, but haven't pinpointed it yet.

lahm86 committed:
change to PS1, remove NB

ajtudela commented:
Done #953
